x/genaccounts/legacy/v0_36/migrate.go

```diff
@@ -0,0 +1,174 @@
+// DONTCOVER
+// nolint
+package v0_36
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	v034distr "github.com/cosmos/cosmos-sdk/x/distribution/legacy/v0_34"
+	v034accounts "github.com/cosmos/cosmos-sdk/x/genaccounts/legacy/v0_34"
+	v034gov "github.com/cosmos/cosmos-sdk/x/gov/legacy/v0_34"
+	v034staking "github.com/cosmos/cosmos-sdk/x/staking/legacy/v0_34"
+
+	"github.com/tendermint/tendermint/crypto"
+)
+
+const (
+	notBondedPoolName = "not_bonded_tokens_pool"
+	bondedPoolName    = "bonded_tokens_pool"
+	feeCollectorName  = "fee_collector"
+	mintModuleName    = "mint"
+
+	basic  = "basic"
+	minter = "minter"
+	burner = "burner"
+)
+
+// Migrate accepts exported genesis state from v0.34 and migrates it to v0.36
+// genesis state. It deletes the governance base accounts and creates the new module accounts.
+// The remaining accounts are updated to the new GenesisAccount type from 0.36
+func Migrate(
+	oldGenState v034accounts.GenesisState, fees sdk.Coins, communityPool sdk.DecCoins,
+	deposits []v034gov.DepositWithMetadata, vals v034staking.Validators, ubds []v034staking.UnbondingDelegation,
+	valOutRewards []v034distr.ValidatorOutstandingRewardsRecord, bondDenom, distrModuleName, govModuleName string,
+) GenesisState {
+
+	depositedCoinsAccAddr := sdk.AccAddress(crypto.AddressHash([]byte("govDepositedCoins")))
+	burnedDepositCoinsAccAddr := sdk.AccAddress(crypto.AddressHash([]byte("govBurnedDepositCoins")))
+
+	bondedAmt := sdk.ZeroInt()
+	notBondedAmt := sdk.ZeroInt()
+
+	// remove the two previous governance base accounts for deposits and burned
+	// coins from rejected proposals add six new module accounts:
+	// distribution, gov, mint, fee collector, bonded and not bonded pool
+	var (
+		newGenState GenesisState
+		govCoins    sdk.Coins
+	)
+
+	for _, acc := range oldGenState {
+		switch {
+		case acc.Address.Equals(depositedCoinsAccAddr):
+			govCoins = acc.Coins
+
+		case acc.Address.Equals(burnedDepositCoinsAccAddr):
+			// do nothing
```

Comment:
Can't this cause problems?

Reply:
No, we need to delete that account. Those coins should have been burned